pkg/controller/container-runtime-config/helpers.go

```diff
@@ -539,24 +539,93 @@ func validateRegistriesConfScopes(insecure, blocked, allowed []string, icspRules
 		}
 	}
 
-	for _, icsp := range icspRules {
-		for _, mirrorSet := range icsp.Spec.RepositoryDigestMirrors {
+	idmsRuleSourcePolicy := map[string]apicfgv1.MirrorSourcePolicy{}
+	itmsRuleSourcePolicy := map[string]apicfgv1.MirrorSourcePolicy{}
+	for _, idms := range idmsRules {
+		for _, mirrorSet := range idms.Spec.ImageDigestMirrors {
+			// keep the field (source, mirror) empty checking for ImageContenSourcePolicy objects
+			// ImageDigestMirrorSet has schema validation
```

Comment:
I don’t understand what this comment is intended to mean.

Reply:
Removed the comment.

I explained the empty string and reject mirror using regex validation are needed for `ImageDigestMirrorSet` that is converted from `ImageContentSourcePolicy`.
normal `ImageDigestMirrorSet` does not need this validation since the API parser does the check. Removed the comment since the ICSP is not processed here.

Reply:
Ah, that is actually a good point that is worth recording. I’m afraid I didn’t make a connection between the two sentences.

Maybe something vaguely like
“Unlike ImageContentSourcePolicy, we don’t need to reject empty string values here because Image{Digest,Tag}MirrorSet rejects such values in schema validation already”.
?